packages/firestore/test/unit/local/test_remote_document_cache.ts

```diff
@@ -67,13 +68,42 @@ export class TestRemoteDocumentCache {
     );
   }
 
-  getNextDocumentChanges(): Promise<MaybeDocumentMap> {
+  getNewDocumentChanges(): Promise<MaybeDocumentMap> {
     return this.persistence.runTransaction(
-      'getNextDocumentChanges',
+      'getNewDocumentChanges',
       'readonly',
       txn => {
         return this.cache.getNewDocumentChanges(txn);
       }
     );
   }
+
+  resetLastProcessDocumentChange(): Promise<void> {
+    return this.persistence.runTransaction(
+      'resetLastProcessDocumentChange',
+      'readonly',
+      txn => {
+        return this.cache.resetLastProcessedDocumentChange(txn);
+      }
+    );
+  }
+
+  removeDocumentChangesThroughChangeId(changeId: number): Promise<void> {
+    if (!(this.cache instanceof IndexedDbRemoteDocumentCache)) {
+      throw new Error(
+        'Can only removeDocumentChangesThroughChangeId() in IndexedDb'
+      );
+    }
+    return this.persistence.runTransaction(
+      'removeDocumentChangesThroughChangeId',
+      'readwrite-primary',
+      txn => {
+        return (this
+          .cache as IndexedDbRemoteDocumentCache).removeDocumentChangesThroughChangeId(
```

Comment:
Is this cast necessary?  If you moved your if check above into this closure it probably wouldn't be...

Reply:
No, I managed to get rid of it by moving the assert.